src/OrchardCore/OrchardCore.Users.Core/Services/UserStore.cs

```diff
@@ -56,6 +57,9 @@ public UserStore(ISession session,
 
     public IEnumerable<IUserEventHandler> Handlers { get; private set; }
 
+    public IQueryable<IUser> Users => _session.Query<User>().ListAsync().GetAwaiter().GetResult()
```

Comment:
At this point this is not longer Queryable code.  You can't promise something you can't accommodate as this case.

Reply:
You can't do this. It will load every user (probably thousands) into memory synchronously. It will break.

In contrast, `Roles` are stored in a single record.

Reply:
```csharp
    public IQuery<User, UserIndex> GetUsers()
    {
        var session = ShellScope.Services.GetService<ISession>();

        return session.Query<User, UserIndex>();
    }
```

Reply:
We can't expose YesSql in the interface, so can't return `IQuery`